stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java

```diff
@@ -61,8 +62,18 @@ public static Object nativeRespond(ObjectValue connectionObj, ObjectValue outbou
         HttpCarbonMessage inboundRequestMsg = HttpUtil.getCarbonMsg(connectionObj, null);
         Strand strand = Scheduler.getStrand();
         DataContext dataContext = new DataContext(strand, new NonBlockingCallback(strand), inboundRequestMsg);
-        HttpCarbonMessage outboundResponseMsg = HttpUtil
-                .getCarbonMsg(outboundResponseObj, HttpUtil.createHttpCarbonMessage(false));
+        HttpCarbonMessage outboundResponseMsg;
+        if (outboundResponseObj.get(RESPONSE_CACHE_CONTROL_FIELD) == null && outboundResponseObj.
```

Comment:
Shall we move this conditional check to a smaller method which has a descriptive method name? So that code will be readable